Blocks/MMQ/mmq.h

```diff
@@ -201,13 +187,16 @@ class MMQImpl {
 
   // Returns { successful allocation flag, circular buffer index }.
   template <bool DROP = DROP_ON_OVERFLOW>
-  typename std::enable_if<DROP, std::pair<bool, size_t>>::type CircularBufferAllocate(
-      std::chrono::microseconds timestamp) {
+  typename std::enable_if<DROP, std::pair<bool, size_t>>::type CircularBufferAllocate(std::chrono::microseconds us) {
     // Implementation that discards the message if the queue is full.
     // MUTEX-LOCKED.
     std::lock_guard<std::mutex> lock(mutex_);
     if (circular_buffer_[head_].status == Entry::FREE) {
       // Regular case.
+      const auto timestamp = us.count() >= 0 ? us : current::time::Now();
```

Comment:
Yes, there is no default parameter. But it doesn't mean the `CircularBufferAllocate` couldn't be called with `us == std::chrono::microseconds(-1)`, it just mean no one needs to call it without specifying `us` value explicitly. The default `-1` value from `EntryPersister::Publish` would be passed here through the `MMQImpl::DoPublish` without modification. And here, at last, we can call `Now()`.